paddle/fluid/framework/ir/memory_optimize_pass/share_mem_opt_info_to_subgraph_pass.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include "paddle/fluid/framework/details/computation_op_handle.h"
+#include "paddle/fluid/framework/details/eager_deletion_op_handle.h"
+#include "paddle/fluid/framework/ir/graph_helper.h"
+#include "paddle/fluid/framework/ir/memory_optimize_pass/memory_optimization_var_info.h"
+#include "paddle/fluid/framework/paddle2cinn/build_cinn_pass.h"
+#include "paddle/fluid/framework/paddle2cinn/cinn_compiler.h"
+#include "paddle/fluid/operators/cinn/cinn_launch_op.h"
+#include "paddle/fluid/platform/enforce.h"
+#include "paddle/fluid/string/string_helper.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+using Name2VarInfoMap =
+    std::unordered_map<std::string, std::shared_ptr<MemOptVarInfo>>;
+
+static details::EagerDeletionOpHandle* FindFollowedEagerDeletionOp(
+    details::ComputationOpHandle* compute_op) {
+  for (auto* var : compute_op->Outputs()) {
+    if (!var->Node()->IsCtrlVar()) {
+      continue;
+    }
+    for (auto* op : var->PendingOps()) {
+      auto* eager_deletion_op =
+          dynamic_cast<details::EagerDeletionOpHandle*>(op);
+      if (eager_deletion_op) {
+        return eager_deletion_op;
+      }
+    }
+  }
+  return nullptr;
+}
+
+static void ShareVarInfoToCinnLaunch(
+    const MemOptVarInfoMapList& var_infos,
+    details::ComputationOpHandle* cinn_launch_op) {
+  auto* followed_eager_deletion_op =
```

Comment:
Will there just be 1 eager_deletion_op or not only 1?

Reply:
every cinn_launch_op only have one at most